node/src/chain/fsm.rs

```diff
@@ -255,18 +256,15 @@ impl<N: Network, DB: database::DB, VM: vm::VMExecution> SimpleFSM<N, DB, VM> {
                         }
                     }
                 }
-                stall_chain_fsm::State::Stalled => {
+                stall_chain_fsm::State::Stalled(_) => {
                     self.blacklisted_blocks.write().await.clear();
                 }
                 _ => {}
             }
+
+            // Ensure that an error in FSM does not affect the stalled_sm
+            fsm_res?;
         }
-        // FIXME: The block should return only if accepted. The current issue is
-        // that the impl of State::on_block_event doesn't return always the
-        // accepted block, so we can't rely on them
-        //
-        // Due to this issue, we reset the outer timeout even if we are not
-        // accepting the received block
```

Comment:
Has this been fixed?

Reply:
The return value is not in use since we deprecated `ACCEPT_TIMEOUT` + on_idle from chain

Reply:
While the issue does not affect the timeout reset anymore, it is still true we sometimes return the block even it it's not accepted.
I agree on removing this comment, but let's update the description of `on_block_event` to state this fact, so we don't assume in the future that this function only returns a block if accepted.

Reply:
Not a blocker (will be addressed in next PR)